concierge_scheduler/concierge_cloud.py

```diff
@@ -0,0 +1,90 @@
+import json
+import os
+import sys
+import logging
+from google.cloud import storage
+from google.oauth2 import service_account
+
+
+class CloudBackupInterface:
+    def __init__(self, bucket_name, credential_file, config_dir, bucket_folder=''):
+        self.config_dir = config_dir
+        self.bucket_name = bucket_name
+        self.bucket_folder = bucket_folder
+        self.credential_file_path = credential_file
+        self.credential_file = None
+        self.config_files = []
+        self.client = None
+        self.__LOG = self.get_logger(__name__)
+        self.command_mapping = {
+            'upload': self.upload_local_dir
+        }
+
+    def run(self, action):
+        self.command_mapping[action]()
+
+    # logging
+    def get_logger(self, name):
+        logger = logging.getLogger(name)
+        logger.setLevel(logging.INFO)
+
+        stream = logging.StreamHandler()
+        fmt = logging.Formatter('%(asctime)s [%(threadName)s] '
+                                '[%(name)s] %(levelname)s: %(message)s')
+        stream.setFormatter(fmt)
+        logger.addHandler(stream)
+
+        return logger
+
+    def _info(self, message, *args):
+        self.__LOG.log(logging.INFO, message.format(*args))
+
+    def _warn(self, message, *args):
+        self.__LOG.log(logging.WARN, message.format(*args))
+
+    def _log_error_and_fail(self, message, *args):
+        self.__LOG.log(logging.ERROR, message.format(*args))
+        sys.exit(-1)
+
+    def authenticate(self):
+        """Load in the credential file for cloud backup"""
+        f = open(self.credential_file_path, 'r')
+        self._info('Authenticating with {} credentials...', self.credential_file_path)
+        try:
+            self.credential_file = json.load(f)
+        except:
+            self._warn('File {} not in JSON format, JSON format required for GCP authentication', self.credential_file_path)
+            self.credential_file = f.read()
+        f.close()
+
+    def upload_local_dir(self):
```

Comment:
I think `upload_local_dir()` needs to be simply `raise NotImplementedError` and if there is a method which collects up local files and directories which is consistent across all implementations, this should be implemented in the interface

Reply:
This now just `raise NotImplementedError` and the method for getting the files is `_get_config_files()` in the interface.